x/crosschain/keeper/cctx_gateway_observers.go

```diff
@@ -32,28 +35,43 @@ InitiateOutbound updates the store so observers can use the PendingCCTX query:
 */
 func (c CCTXGatewayObservers) InitiateOutbound(
 	ctx sdk.Context,
-	cctx *types.CrossChainTx,
-) (newCCTXStatus types.CctxStatus) {
+	config InitiateOutboundConfig,
+) (newCCTXStatus types.CctxStatus, err error) {
 	tmpCtx, commit := ctx.CacheContext()
-	outboundReceiverChainID := cctx.GetCurrentOutboundParam().ReceiverChainId
-	err := func() error {
-		err := c.crosschainKeeper.PayGasAndUpdateCctx(
-			tmpCtx,
-			outboundReceiverChainID,
-			cctx,
-			cctx.InboundParams.Amount,
-			false,
-		)
-		if err != nil {
-			return err
+	outboundReceiverChainID := config.CCTX.GetCurrentOutboundParam().ReceiverChainId
+	// TODO: does this condition make sense?
+	noEthereumTxEvent := false
+	if chains.IsZetaChain(config.CCTX.InboundParams.SenderChainId) {
+		noEthereumTxEvent = true
+	}
```

Comment:
I see.
We introduced this boolean because emitting the events for `CallEVM` done inside the `PostTx` hook had a bug (the list of events got corrupted).
For more information: https://github.com/zeta-chain/node/issues/1010

The NoTx should be used if the function is called from PostTx hook (which is the case when source chain is ZetaChain)

Also if you want to have a look at the issue above feel free. We deprioritized investigating more the issue in the past